config/filter.d/nextcloud-auth.conf

```diff
@@ -0,0 +1,18 @@
+# Fail2Ban filter file for Nextcloud login failures
+#
+# Author: Eric Wolf
+#
+
+[INCLUDES]
+
+# Read common prefixes
+before = common.conf
+
+after = nextcloud-auth.local
+
+[Definition]
+
+# based on https://docs.nextcloud.com/server/27/admin_manual/installation/harden_server.html#setup-a-filter-and-a-jail-for-nextcloud
+_groupsre = (?:(?:,?\s*"\w+":(?:"[^"]+"|\w+))*)
+failregex = ^%(__prefix_line)s?\{%(_groupsre)s,?\s*"remoteAddr":"<HOST>"%(_groupsre)s,?\s*"message":"Login failed:
+datepattern = ,?\s*"time"\s*:\s*"%%Y-%%m-%%d[T ]%%H:%%M:%%S(%%z)?"
```

Comment:
I think the different formats are from a predecessor of Nextcloud called ownCloud and the versions before 18 which apparently had a different timestamp.
The pull request which added it to the hardening guide (see my comment on the author) mentions it working on Nextcloud 18.
Since it still works on Nextcloud 28 and the filter is used by [NextcloudPi](https://github.com/nextcloud/nextcloudpi/blob/f64af68384a359a7c008deb2ef2ab8940f15ba84/bin/ncp/SECURITY/fail2ban.sh#L63C1-L71C14) it seems that it is still valid.

I anchored the time pattern by prefixing it with `^%(__prefix_line)s?\{%(_groupsre)s`.